components/layout_thread/lib.rs

```diff
@@ -1339,6 +1339,17 @@ impl LayoutThread {
             }
         }
 
+        debug!(
+            "Shadow roots in document {:?}",
+            document.shadow_roots().len()
+        );
+
+        let device = Device::new(MediaType::screen(), initial_viewport, device_pixel_ratio);
+        // Flush shadow roots stylesheets if dirty.
+        for shadow_root in document.shadow_roots() {
```

Comment:
Doing this in each layout update is unnecessary, can you file a followup to ensure we only do this at least if any shadow root needs the stylesheets flushed?

Reply:
I implementing this in https://github.com/servo/servo/pull/22743/commits/66f84025696c920bbba4648b569ec9fbc4dfb590